domain/relation/state/relation.go

```diff
@@ -230,6 +231,114 @@ WHERE  relation_uuid = $relationUUID.uuid
 	return relationEndpoints, nil
 }
 
+// GetRegularRelationUUIDByEndpointIdentifiers gets the UUID of a regular
+// relation specified by two endpoint identifiers.
+//
+// The following error types can be expected to be returned:
+//   - [relationerrors.RelationNotFound] is returned if endpoints cannot be
+//     found.
+func (st *State) GetRegularRelationUUIDByEndpointIdentifiers(
+	ctx context.Context,
+	endpoint1, endpoint2 internalrelation.EndpointIdentifier,
+) (corerelation.UUID, error) {
+	db, err := st.DB()
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	type endpointIdentifier1 endpointIdentifier
+	type endpointIdentifier2 endpointIdentifier
+	e1 := endpointIdentifier1{
+		ApplicationName: endpoint1.ApplicationName,
+		EndpointName:    endpoint1.EndpointName,
+	}
+	e2 := endpointIdentifier2{
+		ApplicationName: endpoint2.ApplicationName,
+		EndpointName:    endpoint2.EndpointName,
+	}
+	uuid := relationUUID{}
+
+	stmt, err := st.Prepare(`
+SELECT &relationUUID.*
+FROM   relation r
+JOIN   v_relation_endpoint_identifier e1 ON r.uuid = e1.relation_uuid
+JOIN   v_relation_endpoint_identifier e2 ON r.uuid = e2.relation_uuid
+WHERE  e1.application_name = $endpointIdentifier1.application_name 
+AND    e1.endpoint_name    = $endpointIdentifier1.endpoint_name
+AND    e2.application_name = $endpointIdentifier2.application_name 
+AND    e2.endpoint_name    = $endpointIdentifier2.endpoint_name
+`, uuid, e1, e2)
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err = tx.Query(ctx, stmt, e1, e2).Get(&uuid)
+		if errors.Is(err, sqlair.ErrNoRows) {
```

Comment:
Check to make sure this is only one relation UUID returned.

Reply:
Added it in, though I'm not sure I understand what value this adds aside from checking we haven't gone into an inconsistent state.